saleor/checkout/utils.py

```diff
@@ -656,29 +667,36 @@ def change_shipping_address_in_checkout(checkout, address):
         checkout.save(update_fields=["shipping_address"])
 
 
-def get_checkout_context(checkout, discounts, currency=None, shipping_range=None):
+def get_checkout_context(
+    checkout: Checkout,
+    discounts: "DiscountsListType",
+    currency: Optional[str] = None,
+    shipping_range=None,
+) -> dict:
     """Retrieve the data shared between views in checkout process."""
-    manager = get_extensions_manager()
-    checkout_total = (
-        manager.calculate_checkout_total(checkout=checkout, discounts=discounts)
-        - checkout.get_total_gift_cards_balance()
-    )
+    cards_balance = checkout.get_total_gift_cards_balance()
```

Comment:
Shouldn't be done in `get_checkout_total`?

Reply:
No. Gift cards are payment method. `checkout_total` is `price of items and handling, minus discounts`, not `how much is left to pay`.
IMO we need to discuss how we calculate 'totals' for checkout and how we handle discounts for them